packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -28,27 +27,55 @@ while getopts "r:" option; do
     esac
 done
 
-if [ -z "$resourceGroupName" ]; then
+if [ -z $resourceGroupName ]; then
     exitWithUsageInfo
 fi
 
+# Will return the function name if we deployed before the function app template on this resource group
+functionAppName=$(az group deployment show -g "$resourceGroupName" -n "function-app-template"  --query "properties.parameters.name.value" -o tsv)
+
+# Boolean to track if we should create a new App Registration or there's one already existing
+createNewApp=true
+ 
+# Check if the Function App name is valid and the Function App exists
+if [ ! -z "$functionAppName" ] && (az functionapp show -n "$functionAppName" -g "$resourceGroupName" 1>/dev/null); then
+    # Get the Client (App) ID for the App Registration that's used for this Function App authentication
+    clientId=$(az webapp auth show -n "$functionAppName" -g "$resourceGroupName" --query "clientId" -o tsv)
+    # Check if the App Registration name is valid and the App Registration exists
+    if [ ! -z "$clientId" ] && (az ad app show --id "$clientId" 1>/dev/null); then
+        # If we hit this line, that means we don't need to create a new App Registration
+        createNewApp=false
+    fi
+fi
+
+if ("$createNewApp" = true); then
+    appRegistrationName="allyappregistration_${resourceGroupName}"
```

Comment:
appRegistrationName shouldn't be tied to the resource group?

Reply:
change it to allyfunctionApp_${environment}_${resourceGroupName}

Reply:
I did it, but I'm not totally sold on it, if we want it to be also a standalone script, so the environment isn't related, maybe the subscription id.